nodes/viz/viewer_nurbs_curve.py

```diff
@@ -0,0 +1,128 @@
+# This file is part of project Sverchok. It's copyrighted by the contributors
+# recorded in the version control history of the file, available from
+# its original location https://github.com/nortikin/sverchok/commit/master
+#  
+# SPDX-License-Identifier: GPL3
+# License-Filename: LICENSE
+
+import numpy as np
+from itertools import zip_longest
+import traceback
+
+import bpy
+from mathutils import Matrix, Vector
+from bpy.props import StringProperty, BoolProperty, IntProperty, EnumProperty
+
+from sverchok.node_tree import SverchCustomTreeNode, throttled
+from sverchok.data_structure import Matrix_generate, match_long_repeat, updateNode, get_data_nesting_level, ensure_nesting_level, describe_data_shape, zip_long_repeat, fullList
+from sverchok.utils.sv_obj_helper import SvObjHelper
+
+class SvNurbsCurveOutNode(bpy.types.Node, SverchCustomTreeNode, SvObjHelper):
+    """
+    Triggers: Output NURBS Curve
+    Tooltip: Create Blender's NURBS Curve object
+    """
+
+    bl_idname = 'SvNurbsCurveOutNode'
+    bl_label = 'NURBS Curve Out'
+    bl_icon = 'META_BALL'
+    sv_icon = 'SV_METABALL'
+
+    data_kind: StringProperty(default='CURVE')
+
+    def get_curve_name(self, index):
+        return f'{self.basedata_name}.{index:04d}'
+
+    def create_curve(self, index):
+        object_name = self.get_curve_name(index)
+        surface_data = bpy.data.curves.new(object_name, 'CURVE')
+        surface_data.dimensions = '3D'
+        surface_object = bpy.data.objects.get(object_name)
+        if not surface_object:
+            surface_object = self.create_object(object_name, index, surface_data)
+        return surface_object
+
+    def find_curve(self, index):
+        object_name = self.get_curve_name(index)
+        return bpy.data.objects.get(object_name)
+
+    is_cyclic : BoolProperty(
+            name = "Cyclic",
+            default = False,
+            update = updateNode)
+
+    use_endpoint : BoolProperty(
+            name = "Endpoint",
+            default = True,
+            update = updateNode)
+
+    degree : IntProperty(
+            name = "Degree",
+            min = 2, max = 6,
+            default = 3,
+            update = updateNode)
+
+    def draw_buttons(self, context, layout):
+        self.draw_live_and_outliner(context, layout)
+        self.draw_object_buttons(context, layout)
+        layout.prop(self, 'is_cyclic', toggle=True)
+        row = layout.row(align=True)
+        row.prop(self, 'use_endpoint', toggle=True)
+        row.enabled = not self.is_cyclic
+
+    def sv_init(self, context):
+        self.inputs.new('SvVerticesSocket', 'ControlPoints')
+        self.inputs.new('SvStringsSocket', 'Weights')
+        self.inputs.new('SvStringsSocket', "Degree").prop_name = 'degree'
+        self.outputs.new('SvObjectSocket', "Objects")
+
+    def process(self):
+        if not self.activate:
+            return
+
```

Comment:
stick a  with self.sv_throttle_tree_update()  here, any part of this process function that directly modifies Blender data, will trigger Sverchok updates otherwise.